app/uk/gov/hmrc/cgtpropertydisposals/service/dms/DmsSubmissionPoller.scala

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposals.service.dms
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.ActorSystem
+import cats.implicits._
+import javax.inject.{Inject, Singleton}
+import uk.gov.hmrc.cgtpropertydisposals.service.DmsSubmissionService
+import uk.gov.hmrc.cgtpropertydisposals.util.Logging
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.bootstrap.config.ServicesConfig
+import uk.gov.hmrc.workitem.{Failed, PermanentlyFailed, Succeeded}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+@Singleton
+class DmsSubmissionPoller @Inject() (
+  actorSystem: ActorSystem,
+  dmsSubmissionService: DmsSubmissionService,
+  dmsSubmissionPollerContext: DmsSubmissionPollerContext,
+  servicesConfig: ServicesConfig
+)(implicit
+  executionContext: ExecutionContext
+) extends Logging {
+
+  private val initialPollerDelay: FiniteDuration =
+    FiniteDuration(servicesConfig.getDuration("dms.submission-poller.initial-delay").toNanos, TimeUnit.NANOSECONDS)
+
+  private val pollerInterval: FiniteDuration =
+    FiniteDuration(servicesConfig.getDuration("dms.submission-poller.interval").toNanos, TimeUnit.NANOSECONDS)
+
+  private val failureCountLimit: Int = servicesConfig.getInt("dms.submission-poller.failure-count-limit")
+
+  val _ = actorSystem.scheduler.schedule(initialPollerDelay, pollerInterval)(poller())(dmsSubmissionPollerContext)
+
+  @SuppressWarnings(Array("org.wartremover.warts.Any"))
+  def poller(): Unit = {
+    val _ = dmsSubmissionService.dequeue.map {
+      case Some(workItem) =>
+        if (workItem.failureCount === failureCountLimit) {
+          val _ = dmsSubmissionService.setResultStatus(workItem.id, PermanentlyFailed)
+          ()
+        } else {
+          logger.info(
+            s"DMS Submission poller: " +
+              s"processing work item ${workItem.toString}"
+          )
+
+          implicit val hc: HeaderCarrier = workItem.item.headerCarrier
```

Comment:
will the bearer token still be valid after a certain amount of time? Isn't this the same problem we encountered when we tried to do retries on failed tax enrolments outside of someone's session?

Reply:
it doesn't look like it's authenticated actually, maybe we're ok: 
https://github.com/hmrc/gform/blob/master/app/uk/gov/hmrc/gform/dms/DmsSubmissionController.scala#L40

Reply:
it should be ok, we just really require the bearer token when making the `gforms` request - wasn't very enthused about storing the header-carrier away but in reality that is what we are doing - suspending the request until some future point in time.

Reply:
hmm...I tried to take the header carrier out which meant removing the implicit that is threaded across the modules which are used to make submissions and I hit an issue with the generation of the temporary file created to make the multipart form request. The session-id in the `hc` is used to as part of the file name in order to ensure that it is unqiue. That can be changed but it is hitting quite a few methods so not a small change. I say we go with this and create a tech debt ticket to resolve this later.

Reply:
ok, let's leave it as it is then  - we can address that later if we need to. It's easier to take stuff out than it is to put things in so we should be good